cmd/bacalhau/docker_run.go

```diff
@@ -315,20 +327,30 @@ var dockerRunCmd = &cobra.Command{
 		if err != nil {
 			return fmt.Errorf("error executing job: %s", err)
 		}
-
-		err = ExecuteJob(ctx,
-			cm,
-			cmd,
-			jobSpec,
-			jobDeal,
-			ODR.IsLocal,
-			ODR.WaitForJobToFinish,
-			ODR.DockerRunDownloadFlags)
-
+		if !ODR.DryRun {
```

Comment:
Don't do this. Just see if it's a DryRun, and if it is, print out the jobspec, and return.